markdown/src/main/java/it/niedermann/android/markdown/MarkdownUtil.java

```diff
@@ -275,29 +275,132 @@ public static int getOrderedListNumber(@NonNull String line) {
+                                            editable.delete(newSelectionEnd , newSelectionEnd+punctuation.length());
+                                        }
+                                        return newSelectionEnd ;
+                                    }
+                                    if (temp.contains(punctuationTriple)){
+                                        final int containedPunctuationCount = getContainedPunctuationCount(editable, 0, temp.length(), punctuation);
+                                        if (containedPunctuationCount%2==1){
+                                            return selectionEnd;
+                                        }
+                                        String tmp[]=temp.split(punctuationRegTriple);
+                                        int newSelectionStart=0;
+                                        int newSelectionEnd=0;
+                                        newSelectionStart=tmp[0].length()+newSelectionEnd+punctuation.length()*2;
+                                        for (int i=0;i<=tmp.length-3;i+=2){
+                                            newSelectionEnd=tmp[i+1].length()+newSelectionStart+punctuation.length()*2;
+                                            editable.delete(newSelectionStart, newSelectionStart+punctuation.length());
+                                            editable.delete(newSelectionEnd , newSelectionEnd+punctuation.length());
+                                            newSelectionStart=tmp[i+2].length()+newSelectionEnd+punctuation.length()*2;
+
+                                        }
+                                        return newSelectionEnd-punctuation.length()*2 ;
+                                    }else {
+
+                                        String tmp[]=temp.split(punctuationRegDouble);
+                                        int newSelectionStart=0;
+                                        int newSelectionEnd=0;
+                                        newSelectionStart=tmp[0].length()+newSelectionEnd+punctuation.length()*2;
+
+                                        for (int i=0;i<=tmp.length-3;i+=2){
+                                            newSelectionEnd=tmp[i+1].length()+newSelectionStart+punctuation.length()*2+(1);
+                                            editable.insert(newSelectionStart, punctuation);
+                                            editable.insert(newSelectionEnd , punctuation);
+                                            newSelectionStart=tmp[i+2].length()+newSelectionEnd+punctuation.length()*2+(1);
+
+
+                                        }
+                                        return newSelectionEnd+1 ;
+                                    }
+
+                            }
+                            break;
+                        case 2:
+                            switch (punctuation){
```

Comment:
I think for the sake of better readability, we can abstain from making another `switch` statement here:

We already switched above and there can only be those three cases. While your implementation is fail safe in case we add more punctuation cases which behave differently, i'd rather like to see the error instead of failing silently :slightly_smiling_face: